### PR TITLE
Not create new XDT file when there is one come with package

### PR DESCRIPTION
**Pull request by shrimpy**
when package come with XDT

```xml
<step title="BackgroundTrace" date="2015-03-16T23:42:33.557" instance="XIAOWU" url="/api/siteextensions/webrootxdttest" method="PUT" Accept="application/json" Expect="100-continue" Host="localhost:62483" x-ms-geo-location="" >
  <step title="Version is null, search latest package by id: webrootxdttest, will not search for unlisted package." date="2015-03-16T23:42:33.562" /><!-- duration: 770ms -->
  <step title="Install package: webrootxdttest." date="2015-03-16T23:42:34.334" >
    <step title="Download site extension: webrootxdttest.4.0.0" date="2015-03-16T23:42:34.336" >
      <step title="Use xdt file from package." date="2015-03-16T23:42:34.785" /><!-- duration: 1ms -->
      <step title="Moving E:\Kudu-Test-Files\KuduApps\TestRunnerSiteXIAOWU-KUDU1\site\wwwroot\applicationHost.xdt to E:\Kudu-Test-Files\KuduApps\TestRunnerSiteXIAOWU-KUDU1\SiteExtensions\webrootxdttest\applicationHost.xdt" date="2015-03-16T23:42:34.788" /><!-- duration: 1ms -->
    </step><!-- duration: 455ms -->
  </step><!-- duration: 463ms -->
  <step title="Update arm settings for webrootxdttest installation. Status: OK" date="2015-03-16T23:42:35.189" /><!-- duration: 20ms -->
</step><!-- duration: 1655ms -->
```

when package doesn`t have a xdt
```xml
<step title="BackgroundTrace" date="2015-03-16T23:42:10.366" instance="XIAOWU" url="/api/siteextensions/SimpleSvc" method="PUT" Accept="application/json" Expect="100-continue" Host="localhost:62483" x-ms-geo-location="" >
  <step title="Search package by id: SimpleSvc and version: 1.0.0, will also search for unlisted package." date="2015-03-16T23:42:10.368" /><!-- duration: 16250ms -->
  <step title="Install package: SimpleSvc." date="2015-03-16T23:42:26.620" >
    <step title="Download site extension: SimpleSvc.1.0.0" date="2015-03-16T23:42:26.621" >
      <step title="Generate xdt file to map application to run from wwwroot when request for /SimpleSvc" date="2015-03-16T23:42:28.984" /><!-- duration: 2ms -->
    </step><!-- duration: 2367ms -->
  </step><!-- duration: 2374ms -->
  <step title="Update arm settings for SimpleSvc installation. Status: OK" date="2015-03-16T23:42:29.675" /><!-- duration: 22ms -->
</step><!-- duration: 19334ms -->

```

### CODE REVIEW

File: Kudu.FunctionalTests/SiteExtensionApiFacts.cs [hunk 1]
```diff
@@ -635,6 +641,20 @@ public class SiteExtensionApiFacts
...
+                result = await responseMessage.Content.ReadAsAsync<SiteExtensionInfo>();
+                Assert.Equal(externalPackageWithXdtId, result.Id);
+                Assert.Equal(externalFeed, result.FeedUrl);
+
+                TestTracer.Trace("GET request to verify package content has been copied to wwwroot");
+                responseMessage = await client.GetAsync(appManager.SiteUrl);
+                responseContent = await responseMessage.Content.ReadAsStringAsync();
+                Assert.NotNull(responseContent);
+                Assert.True(responseContent.Contains(@"1 files"));
+                Assert.True(responseContent.Contains(@"site\path\shall\not\be\found"));
```
Comment (shrimpy) [inline Kudu.FunctionalTests/SiteExtensionApiFacts.cs]: "site\path\shall\not\be\found" is the content from xdt file

File: Kudu.Core/SiteExtensions/SiteExtensionManager.cs [hunk 2]
```diff
@@ -456,12 +456,33 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
...
-
-                        tracer.Trace("Generate xdt file to map application to run from wwwroot when request for /{0}", package.Identity.Id);
-                        GenerateRemapToWebRootApplicationHostXdt(installationDirectory, string.Format(CultureInfo.InvariantCulture, "/{0}", package.Identity.Id));
                     }
 
                     await remoteRepo.DownloadPackageToFolder(package.Identity, extractPath, pathToLocalCopyOfNudpk: packageLocalFilePath);
+
+                    if (SiteExtensionInfo.SiteExtensionType.WebRoot == type)
+                    {
+                        // if install to WebRoot, check if there is any xdf file come with package
```
Comment (davidebbo) [inline Kudu.Core/SiteExtensions/SiteExtensionManager.cs]: 'xdf' typo

Comment (shrimpy) [inline Kudu.Core/SiteExtensions/SiteExtensionManager.cs]: oh.. fixed. Thanks!

File: Kudu.Core/SiteExtensions/SiteExtensionManager.cs [hunk 3]
```diff
@@ -456,12 +456,33 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
...
+                        // if there is one move it to site extension folder
+                        // if not, generate one to map request to wwwroot
+                        var xdtFiles = FileSystemHelpers.GetFiles(extractPath, "*.xdt");
```
Comment (davidebbo) [inline Kudu.Core/SiteExtensions/SiteExtensionManager.cs]: We really only support `applicationHost.xdt`, and no others. So may as well make code specific to that, and not need any looping. You can use the `_applicationHostFile` constant.

File: Kudu.Core/SiteExtensions/SiteExtensionManager.cs [hunk 4]
```diff
@@ -456,12 +456,32 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
...
+                        string xdtFile = Path.Combine(extractPath, _applicationHostFile);
+                        if (File.Exists(xdtFile))
+                        {
+                            tracer.Trace("Use xdt file from package.");
+                            FileInfo xdtFileInfo = new FileInfo(xdtFile);
+                            string newXdtFile = Path.Combine(installationDirectory, xdtFileInfo.Name);
+
+                            tracer.Trace("Moving {0} to {1}", xdtFile, newXdtFile);
+                            FileSystemHelpers.MoveFile(xdtFile, newXdtFile);
+
```
Comment (shrimpy) [inline Kudu.Core/SiteExtensions/SiteExtensionManager.cs]: removed empty line

File: Kudu.Core/SiteExtensions/SiteExtensionManager.cs [hunk 5]
```diff
@@ -456,12 +456,32 @@ private SiteExtensionInfo GetPreInstalledExtension(string id)
...
+                    if (SiteExtensionInfo.SiteExtensionType.WebRoot == type)
+                    {
+                        // if install to WebRoot, check if there is any xdt file come with package
+                        // if there is one, move it to site extension folder
+                        // if not, generate one to map request to wwwroot
+                        string xdtFile = Path.Combine(extractPath, _applicationHostFile);
+                        if (File.Exists(xdtFile))
+                        {
+                            tracer.Trace("Use xdt file from package.");
+                            FileInfo xdtFileInfo = new FileInfo(xdtFile);
```
Comment (davidebbo) [inline Kudu.Core/SiteExtensions/SiteExtensionManager.cs]: `var` in this case :)

Comment (davidebbo) [inline Kudu.Core/SiteExtensions/SiteExtensionManager.cs]: But why do you need FileInfo at all here? Why not just `Path.Combine(installationDirectory, _applicationHostFile)`?

**Comment by shrimpy** [comment]
merge in 45741c18a3dae7025e7d465793c99ba951ae141e

**Pull request closed without merging**